### PR TITLE
Time table view

### PR DESCRIPTION
**Pull request by pavitgopal**

### CODE REVIEW

File: app/assets/stylesheets/time_slots.css [hunk 1]
```diff
@@ -0,0 +1,100 @@
+.table-wrapper {
+    max-width: 800px; 
+    margin: 0 auto;   
+    padding: 20px;    
+    background-color: #f8f8f8; 
+    border-radius: 10px;
+    box-shadow: 0 0 10px rgba(0, 0, 0, 0.1); 
+  }
+  
+  table {
```
Comment (wahibkapdi) [inline app/assets/stylesheets/time_slots.css]: User css classes for this. Directly editing a component may hurt someone else's css.

File: app/assets/stylesheets/time_slots.css [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    width: 100%;       
+    border-collapse: collapse;
+  }
+  
+  th, td {
```
Comment (wahibkapdi) [inline app/assets/stylesheets/time_slots.css]: Same as above

File: app/assets/stylesheets/time_slots.css [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    align-items: flex-start;
+  }
+  
+  .filters label {
+    font-weight: bold;
+    margin-bottom: 5px;
+    display: block;
+  }
+  
+  .btn {
```
Comment (wahibkapdi) [inline app/assets/stylesheets/time_slots.css]: This will conflict with the bootstrap call for btn.

File: app/assets/stylesheets/time_slots.css [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    color: white;
+    text-align: center;
+    border-radius: 5px;
+    cursor: pointer;
+    border: none;
+    margin-top: auto; 
+    transition: background-color 0.3s ease;
+  }
+  
+  .btn:hover {
```
Comment (wahibkapdi) [inline app/assets/stylesheets/time_slots.css]: Same as above.

**Review by wahibkapdi — CHANGES_REQUESTED** [review]

**Comment by Naviunni** [comment]
Please add some cucumber tests as well

File: spec/controllers/time_slot_controllers_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+RSpec.describe TimeSlotsController, type: :controller do
+  before do
+    # Create a user and set session user_id as done in your application_controller_spec.rb
+    @user = User.create!(uid: '12345', provider: 'google_oauth2', email: 'test@example.com', first_name: 'John',
+                         last_name: 'Doe')
+    session[:user_id] = @user.id
+  end
+
+  let!(:lec_slot) { TimeSlot.create!(day: 'MWF', start_time: '08:00', end_time: '08:50', slot_type: 'LEC') }
```
Comment (Naviunni) [inline spec/controllers/time_slot_controllers_spec.rb:13]: Probably easier to use the factory to create test objects

File: app/models/time_slot.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+class TimeSlot < ApplicationRecord
+    
+  end
```
Comment (Naviunni) [inline app/models/time_slot.rb:15]: Would be good to add some validations like asserting the presence of start_time, end_time etc

**Pull request closed without merging**